runtime/caml/sync.h

```diff
@@ -76,15 +76,18 @@ Caml_inline int sync_condvar_wait(sync_condvar c, sync_mutex m)
 
 Caml_inline void sync_check_error(int retcode, char * msg)
 {
```

Comment:
Note to self: why is this (big) function inlined and declared in a .h while it's obviously not performance critical and could be static to sync.c ?  Something needs to be done.

Reply:
Looks like the check_sync_error is used in more places than just sync.c, so it looks better to sort this out in a followup PR.

```
otherlibs/systhreads/st_posix.h
388:  sync_check_error(retcode, "Thread.sigmask");
408:  sync_check_error(retcode, "Thread.wait_signal");

otherlibs/systhreads/st_stubs.c
587:    sync_check_error(err, "Thread.create");
592:    sync_check_error(err, "Thread.create");
638:    sync_check_error(err, "caml_register_c_thread");
729:  sync_check_error(rc, "Thread.join");
764:  sync_check_error(st_event_create(&ts), "Thread.create");

runtime/sync.c
100:  sync_check_error(sync_mutex_create(&mut), "Mutex.create");
120:  sync_check_error(retcode, "Mutex.lock");
130:  sync_check_error(retcode, "Mutex.unlock");
140:  sync_check_error(retcode, "Mutex.try_lock");
199:  sync_check_error(sync_condvar_create(&cond), "Condition.create");
218:  sync_check_error(retcode, "Condition.wait");
226:  sync_check_error(sync_condvar_signal(Condition_val(wrapper)),
233:  sync_check_error(sync_condvar_broadcast(Condition_val(wrapper)),

runtime/caml/sync.h
77:Caml_inline void sync_check_error(int retcode, char * msg)
```